### PR TITLE
examples: add FDA training example (pytorch)

### PR DESCRIPTION
**Pull request by saud5150**
Part of #1956

## Description
- [ ] My change is breaking

Adds `examples/pytorch/fda.py` following `examples/pytorch/simclr.py` as a
template. The model and loss are identical to SimCLR (`NTXentLoss` +
`SimCLRProjectionHead`); only the transform differs. `FDATransform` does not
accept `input_size` directly, so view transforms are passed explicitly — same
pattern as `byol.py` and `tico.py`.

## Tests
- [x] My change is covered by existing tests.
- [ ] My change needs new tests.
- [ ] I have added/adapted the tests accordingly.
- [x] I have manually tested the change.

```
make format-check  # All checks passed
make lint          # All checks passed
```
Ran the example on CIFAR-10; loss decreased over 4 epochs.

## Documentation
- [ ] I have added docstrings to all public functions/methods.
- [ ] My change requires a change to the documentation (`.rst` files).
- [ ] I have updated the documentation accordingly.
- [x] The autodocs update the documentation accordingly.

### CODE REVIEW

**Comment by gabrielfruet** [comment]
Thank you for the contribution!

**Comment by gabrielfruet** [comment]
LGTM ! Thank you for the nice contribution

**Review by gabrielfruet — APPROVED** [review]
